server/src/sas/CodeZoneManager.ts

```diff
@@ -831,7 +831,11 @@ export class CodeZoneManager {
         }
       }
       if (token.text[0] === "%") {
-        return CodeZoneManager.ZONE_TYPE.MACRO_STMT; //TODO: need to differ among ARM macro, autocall macro, macro function, macro statement
+        if (token.text.toUpperCase() === "%MACRO") {
```

Comment:
I noticed that there's a side effect to this.
If I type `%macro;`, the autocomplete list is not close when I type `;`, if I try to type Enter to close the autocomplete, the typed `;` will be removed.
I would suggest we don't change this for now. #437 is mainly about uppercase code. For macro, in normal cases it should has a macro name which works fine.

Reply:
OK, reverted the change.